File: BackEnd_Learn/Java/jwt/src/main/java/com/example/jwt/Controller/AuthController.java
package com.example.jwt.Controller;

import com.example.jwt.Payload.Request.LoginRequest;
import com.example.jwt.Payload.Response.TokenResponse;
import com.example.jwt.Service.Auth.AuthService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/auth")
public class AuthController {

    private final AuthService authService;

    @PostMapping
    public TokenResponse auth(@RequestBody LoginRequest request){
        return authService.signIn(request);
    }

    @PutMapping
    public TokenResponse update(@RequestHeader("refresh_token") String request){
        return authService.refreshToken(request);
    }

}

File: BackEnd_Learn/NestJS/nest-jwt/src/auth/auth.controller.ts
import {Body, Controller, Post} from '@nestjs/common';
import {AuthService} from "./auth.service";
import {UserRequest} from './dto/request/user.dto';
import {TokenResponse} from './dto/response/token.dto';

@Controller('auth')
export class AuthController {
    constructor(private readonly authService: AuthService) {}

    @Post()
    async login(@Body() userRequest: UserRequest): Promise<TokenResponse> {
        return this.authService.login(userRequest)
    }

}
